hibernate-core/src/main/java/org/hibernate/boot/model/internal/ClassPropertyHolder.java

```diff
@@ -363,9 +363,14 @@ public void doSecondPass(Map persistentClasses) throws MappingException {
 						else {
 							final Iterator<Property> propertyIterator = component.getPropertyIterator();
 							while ( propertyIterator.hasNext() ) {
-								Property property = propertyIterator.next();
+								final Property property = propertyIterator.next();
 								try {
-									property.getGetter( componentClass );
+									if ( componentClass == Object.class ) {
```

Comment:
What if `MyNonVisibleGenericMappedSuperclass` declares the type parameter as i.e. `C extends Serializable`? Wouldn't the `componentClass` be `Serializable` then and this fails again?

Reply:
do you mean like this https://github.com/hibernate/hibernate-orm/pull/7150/files#diff-eccd008b6d30c86b44553b164402de547e957ddf9596d8f8458ba1d6f05fd546R99 ?

Reply:
Looks about right, but not sure if `MyMappedSuperclass` has to be a top level class like `MyNonVisibleGenericMappedSuperclass` due to the comment of that file.

```
// This class must not be nested in the test class, otherwise its private fields will be visible
// from subclasses and we won't reproduce the bug.
```

Reply:
@beikov add https://github.com/hibernate/hibernate-orm/pull/7150/files#diff-3d3588a6cdbb3ea905616ae5179c0bc6490ada46e44dbe91f6a4ab503055e72f

Reply:
Maybe we could just add the `IllegalArgumentException` type to the catch block, like we already do for `PropertyNotFoundException`? It's a bit generic but it would be consistent with the "if we cannot get this property from this class, remove it from the list" logic.

Reply:
>Maybe we could just add the IllegalArgumentException type to the catch block, like we already do for PropertyNotFoundException? It's a bit generic but it would be consistent with the "if we cannot get this property from this class, remove it from the list" logic.

If the `Object.class` code has problems or doesn't work, then I think reacting on the exception is our only way.

Reply:
I prefer to not rely on a thrown exception in the first place, unless there is no other way. Whatever we do, please add a code comment e.g. `// Object is not a valid component class, but that is what we get when using a type variable`

Reply:
@beikov comment added